convert/lib/nuke.py

```diff
@@ -0,0 +1,434 @@
+                    if association_id:
+                        print(
+                            self.ec2.disassociate_address(AssociationId=association_id)
+                        )
+                    print(self.ec2.release_address(AllocationId=allocation_id))
+
+    def flowlog(self, flow_logs: list[str]):
+        if not flow_logs:
+            return
+        result = self.ec2.describe_flow_logs(FlowLogIds=flow_logs)
+        existing_flow_logs = [i["FlowLogId"] for i in result["FlowLogs"]]
+
+        if existing_flow_logs:
+            pprint({"Flow Log IDs to delete": existing_flow_logs})
+
+            if self.delete:
+                self.ec2.delete_flow_logs(FlowLogIds=existing_flow_logs)
+
+    def instance(self, instance_ids: list[str]):
+        if not instance_ids:
+            return
+        result = self.ec2.describe_instances(InstanceIds=instance_ids)
+        if not result["Reservations"]:
+            return
+        existing_instances = [
+            i["InstanceId"]
+            for i in result["Reservations"][0]["Instances"]
+            if i["State"]["Name"] != "terminated"
+        ]
+
+        if existing_instances:
+            pprint({"Instance IDs to delete": existing_instances})
+
+            if self.delete:
+                self.ec2.terminate_instances(InstanceIds=existing_instances)
+
+    def launch_template(self, launch_templates: list[str]):
+        if not launch_templates:
+            return
+        existing_templates = self._get_from_paged_list(
+            launch_templates,
+            self.ec2.describe_launch_templates,
+            lambda x: [i["LaunchTemplateId"] for i in x["LaunchTemplates"]],
+            "NextToken",
+        )
+
+        if existing_templates:
+            pprint({"Launch Template IDs to delete": existing_templates})
+
+            if self.delete:
+                for template_id in existing_templates:
+                    self.ec2.delete_launch_template(LaunchTemplateId=template_id)
+
+    def security_group(self, security_groups: list[str]):
+        if not security_groups:
+            return
+        result = self.ec2.describe_security_groups()
+        existing_sgs = [
+            i["GroupId"]
+            for i in result["SecurityGroups"]
+            if i["GroupId"] in security_groups
+        ]
+
+        if existing_sgs:
+            pprint({"Security Group IDs to delete": existing_sgs})
+
+            if self.delete:
+                for sg in existing_sgs:
+                    self.ec2.delete_security_group(GroupId=sg)
+
+    def instance_profile(self, instance_profiles: list[str]):
+        if not instance_profiles:
+            return
+        existing_profiles = self._get_from_paged_list(
+            instance_profiles,
+            self.iam.list_instance_profiles,
+            lambda x: [p["InstanceProfileName"] for p in x["InstanceProfiles"]],
+        )
+
+        if existing_profiles:
```

Comment:
Pretty nitpicky, but I like this pattern since it keeps code from extending rightwards:

```
if not existing_profiles:
  return

pprint()

if not self.delete:
  return

...
```

Reply:
It doesn't go very far right, while that pattern would add four lines to every type of thing we nuke.